Add aliases config for command abbreviations

pkratoch opened the pull request:
Not included by default.

Conan-Kudo commented on etc/dnf/aliases.d/command-abbreviations.conf:
```diff
@@ -0,0 +1,19 @@
+se = search
+sh = shell
+up = upgrade
+um-m = upgrade-minimal
```
Wasn't this `up-m`?

Kratochvilova replied:
Almost, it was ``u-m``. Thanks for catching that.

Conan-Kudo commented on etc/dnf/aliases.d/command-abbreviations.conf:
```diff
@@ -0,0 +1,19 @@
+sh = shell
+up = upgrade
+um-m = upgrade-minimal
+um-min = upgrade-minimal
```
`up-min`?

Kratochvilova replied:
Fixed, thanks.

Conan-Kudo commented:
@Kratochvilova I still think you guys are being paranoid to not ship these by default, but whatever... It's a good start.

j-mracek commented:
@Conan-Kudo I am paranoid. The plugin has some downsides. If I want to create an alias for u-m = upgrade-minimal, then command like "dnf --repo u-m upgrade" will result in command "dnf --repo upgrade-minimal upgrade". I am still thinking about an improvement.

Conan-Kudo commented:
@j-mracek I still think the abbreviations were better as actually part of the commands like they were before. That avoids a lot of the issues you're talking about and makes it so that bash-completion can be adapted for it. In minimal environments without bash-completion, the short syntax makes it *much* easier and quicker to use.

j-mracek commented on etc/dnf/aliases.d/command-abbreviations.conf:
```diff
@@ -0,0 +1,19 @@
+[aliases]
+dup = distro-sync
+dsync = distro-sync
+dist-upgrade = distro-sync
```
Do we really want to support 3 abbreviations for one command? I would prefer only one (this is the part that I dislike in formal PR)

Conan-Kudo replied:
`dup` and `dist-upgrade` are compatibility with `zypper` and `apt`. `dsync` is the short form for the dnf command itself.

j-mracek commented on etc/dnf/aliases.d/command-abbreviations.conf:
```diff
@@ -0,0 +1,19 @@
+up = upgrade
+u-m = upgrade-minimal
+up-min = upgrade-minimal
+um = upgrade-minimal
```
Same like for distro-sync

Conan-Kudo replied:
Let's just keep `up-min` for `upgrade-minimal`.

dmnks replied:
@pkratoch hey Pavla, would you please consider the change proposed by @Conan-Kudo above?

pkratoch replied:
Sorry, done.

pkratoch commented:
@j-mracek, so what about this PR? I found out even some Fedora users are now missing some aliases ( https://bugzilla.redhat.com/show_bug.cgi?id=1634232 ).

You mentioned, that you would be ok with adding one alias for each command directly to the code. So how about we add one for each command to the code, and keep the rest here (remaining aliases for distro-sync, makecache, and reinstall)? That way, at least some users won't need to enable the aliases.

dmnks commented:
> You mentioned, that you would be ok with adding one alias for each command directly to the code. So how about we add one for each command to the code, and keep the rest here (remaining aliases for distro-sync, makecache, and reinstall)? That way, at least some users won't need to enable the aliases.

Wouldn't that just make things more scattered? I mean, we would then have two types of aliases that are treated differently (hard-coded vs. configured). Couldn't we ship the few essential aliases in a separate config file perhaps?

pkratoch commented:
> Wouldn't that just make things more scattered? I mean, we would then have two types of aliases that are treated differently (hard-coded vs. configured). Couldn't we ship the few essential aliases in a separate config file perhaps?

We already have some hard-coded aliases, so I don't think it would made much difference. However, I am fine with either way.

Conan-Kudo commented:
@pkratoch I'd prefer for a set of them to be wired into the code myself...

dmnks commented:
> We already have some hard-coded aliases, so I don't think it would made much difference. However, I am fine with either way.

Ah, OK you're right. In that case I'm fine with either as well.

pkratoch commented:
I added one abbreviations for each command, the rest remains in the aliases config. I also added aliases based on the discussion in PR: https://github.com/rpm-software-management/dnf/pull/980

dmnks commented:
FYI, we decided to give this a second look, to see if we can unify the aliases somehow (so that we don't end up with two places where aliases can be defined). More details to follow, I guess.

Conan-Kudo commented:
@dmnks @pkratoch @j-mracek Can we land this soon? I'd like to get this in before the versions freeze for Mageia 7 on April 23 so that we keep functional parity with DNF 2.x in Mageia 6.

pkratoch commented:
@Conan-Kudo I believe all of the commands in the "command-abbreviations.conf" match zypper commands, so is it ok to rename it to "zypper.conf"? After that, I've been told it's ready to be merged.

dmnks commented:
Yeah, while the built-in aliases being added here (such `up` or `dg`) are useful in general, the ones that we're pushing into the `aliases.d` directory are mostly meant to serve `zypper` users, hence the proposed name `zypper.conf` :)

Conan-Kudo commented:
@pkratoch @dmnks I don't have a problem with that.
